cdm/src/main/java/ucar/nc2/NetcdfFile.java

```diff
@@ -114,6 +114,8 @@
     }
     try {
       registerIOProvider("ucar.nc2.iosp.hdf5.H5iosp");
+      if(System.getProperty("Netcdf4Clibrary.useForReading") != null)
+	registerIOProvider("ucar.nc2.jni.netcdf.Nc4Iosp"); // Make sure it precedes hdf5 IOSP
```

Comment:
A couple notes:
* For parity with the rest of the IOSP registrations, this should be in its own try-catch block, not `H5iosp`'s.
* If you follow the logic, this will register `Nc4Iosp` *after* `N5iosp`. I understand why you were confused: we eventually delegate to [registerIOProvider(Class iospClass, boolean last)](https://github.com/Unidata/thredds/blob/v4.6.6/cdm/src/main/java/ucar/nc2/NetcdfFile.java#L307) with `last == false`, but pay attention to the `userLoads` flag.